src/vs/workbench/contrib/notebook/common/notebookEditorModel.ts

```diff
@@ -47,13 +49,29 @@ export class SimpleNotebookEditorModel extends EditorModel implements INotebookE
 		readonly viewType: string,
 		private readonly _workingCopyManager: IFileWorkingCopyManager<NotebookFileWorkingCopyModel, NotebookFileWorkingCopyModel>,
 		@ILifecycleService lifecycleService: ILifecycleService,
-		@IFilesConfigurationService private readonly _filesConfigurationService: IFilesConfigurationService
+		@IFilesConfigurationService private readonly _filesConfigurationService: IFilesConfigurationService,
+		@IExtensionService extensionService: IExtensionService
 	) {
 		super();
 
 		if (this.viewType === 'interactive') {
 			lifecycleService.onBeforeShutdown(async e => e.veto(this.onBeforeShutdown(), 'veto.InteractiveWindow'));
 		}
+
+		this._register(extensionService.onWillStop(e => {
```

Comment:
@rebornix I am not sure you think there is a better place for this. An alternative would be to put it somewhere where all notebook models are known and iterate over all similar to what Matt has done here:

https://github.com/microsoft/vscode/blob/4f86d3c466d3322b768267b23d7ac3e0c97c81a7/src/vs/workbench/api/browser/mainThreadCustomEditors.ts#L110-L128